leasing/models/rent.py

```diff
@@ -1476,6 +1476,118 @@ class LegacyIndex(models.Model):
+    # Maximum lengths are arbitrary, but set to avoid extra large input.
+    CHARFIELD_MAX_LENGTH = 255
+
+    code = models.CharField(
+        verbose_name=_("Code"),
+        max_length=CHARFIELD_MAX_LENGTH,
+    )
+    name = models.CharField(
+        verbose_name=_("Name"),
+        max_length=CHARFIELD_MAX_LENGTH,
+    )
+    comment = models.TextField(verbose_name=_("Region"), blank=True)
+    source = models.CharField(
+        verbose_name=_("Data source"),
+        blank=True,
+        max_length=CHARFIELD_MAX_LENGTH,
+    )
+    source_table_updated = models.DateTimeField(
+        verbose_name=_("Source table updated"), null=True
+    )
+    source_table_label = models.TextField(
+        verbose_name=_("Source table label"),
+        blank=True,
+    )
+    url = models.CharField(
+        verbose_name=_("URL"),
+        max_length=CHARFIELD_MAX_LENGTH,
+    )
+
+    class Meta:
+        verbose_name = pgettext_lazy(
+            "model name", "price index of old dwellings in housing companies"
+        )
+        verbose_name_plural = pgettext_lazy(
+            "model name", "price indexes of old dwellings in housing companies"
+        )
+        constraints = [
+            models.UniqueConstraint(fields=["code", "url"], name="unique_price_index")
+        ]
+
+
+class IndexNumber(models.Model):
+    """
+    In Finnish: Indeksipisteluku
+
+    Holds the index numbers.
+    Currently only used with the newer price indexes related to Level Check (Tasotarkistus).
+
+    Members:
+        index: Reference to the index this number is for.
+        number: Index number. Example: 101.5
+        year: Year for the number. Example: 2020.
+        quarter: Quarter number. Example: 4
+        region: Area for index value. Example: "pks" for Pääkaupunkiseutu / Greater Helsinki area
+        comment: Comment for the number. Example: "* ennakkotieto\r\n"
+    """
+
+    index = models.ForeignKey(
+        OldDwellingsInHousingCompaniesPriceIndex,
+        verbose_name=_("Index number"),
+        on_delete=models.PROTECT,
+        related_name="numbers",
+    )
+    # max_digits is arbitrary for the number. No need to limit it, although 7
+    # should be enough if the numbers are at most in the 100s of thousands.
+    # Largest index number in the system at the moment is year 1914's index
+    # with a number around 260 000.
+    number = models.DecimalField(
+        verbose_name=_("Number"),
+        decimal_places=1,
+        max_digits=8,
+        null=True,
+    )
+    year = models.PositiveSmallIntegerField(verbose_name=_("Year"))
+    quarter = models.PositiveSmallIntegerField(
+        verbose_name=_("Quarter"),
+        validators=[MinValueValidator(1), MaxValueValidator(4)],
+        null=True,
+    )
+    # Max lengths here are arbitrary, but set to avoid extra large input.
+    region = models.CharField(verbose_name=_("Region"), blank=True, max_length=255)
+    comment = models.TextField(verbose_name=_("Region"), blank=True)
+
```

Comment:
Since you asked, I would prefer to have `updated_at` field at least. Could be useful - if there ever is a need to figure out issues. E.g. in 2035 the value for 2025Q1 is updated.

Reply:
I have understood that the `updated_at` field will be updated every time the `update_or_create` function is called, regardless if any field values were changed or not. This would make it only and indicator when the importer was last run, and not when the field values were changed. Or does the timestamp work differently?

Reply:
Added `updated_at` and `created_at` as part of the `TimeStampedModel`.